2015/04/Upgrade_Insecure.md

```diff
@@ -0,0 +1,102 @@
+# Upgrade Insecure Requests Draft Feedback
+
+[Draft under discussion.](https://w3c.github.io/webappsec/specs/upgrade/) ([At this revision](https://github.com/w3c/webappsec/tree/f3ef3fb43394d57fa3ff9f93f22f3b057f7551be/specs/upgrade))
+
+## General Discussion
+
+This draft defines a new Content Security Policy header for a page to assert
+that subresources should be automatically upgraded to their secure version. We
+think this is a great step towards reducing barriers that website publishers
+face when transitioning from plaintext HTTP to secure connections.
+
+### ISSUE: Goal 1 Unclear
+
+Section 1.1, Goal 1:
+
+> Authors should be able to ensure that all content requested by a given page
+> loads successfully, and securely. Mixed content blocking should not break
+> pages as a result of migrating to a secure origin.
+
+This seems somewhat too ambitious for the spec. If third-party content on
+a page does not support HTTPS or stops supporting HTTPS, the page author cannot
+ensure that the content is loaded securely or at all. Inevitably, moving to
+a secure origin causes problems with mixed content blocking if the page has
+third party content that doesn't yet support HTTPS, a problem which the spec
+does not address.
```

Comment:
It's meant to be read in context of the preceding paragraph of conditions: "If we assume that ... authors also ensure that content is accessible at the same host and path on a secure scheme, then ...".